pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1176,9 +1171,31 @@ func (n *NetworkProvider) reconcileStandaloneVPCSubnets(ctx context.Context, log
 	return subs, nil
 }
 
-// getVPCRouteTable will return a list of route tables based on a route table tag
+func (n *NetworkProvider) getClusterRouteTables(ctx context.Context) ([]*ec2.RouteTable, error) {
+	routeTables, err := n.Ec2Api.DescribeRouteTables(&ec2.DescribeRouteTablesInput{})
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get route tables")
+	}
+
+	clusterVPC, err := getClusterVpc(ctx, n.Client, n.Ec2Api, n.Logger)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get cluster vpc")
+	}
+	var foundRouteTables []*ec2.RouteTable
+	for _, routeTable := range routeTables.RouteTables {
+		if *routeTable.VpcId == *clusterVPC.VpcId {
+			foundRouteTables = append(foundRouteTables, routeTable)
+		}
+	}
+	if len(foundRouteTables) == 0 {
+		return nil, errorUtil.New(fmt.Sprintf("could not find any route table with the associated vpc id %s", aws.StringValue(clusterVPC.VpcId)))
+	}
+	return foundRouteTables, nil
+}
+
+// getVPCRouteTableByTag will return a list of route tables based on a route table tag
 // we expect there to be route tables, if none are found we return an error
-func (n *NetworkProvider) getVPCRouteTable(routeTableTag *ec2.Tag) ([]*ec2.RouteTable, error) {
+func (n *NetworkProvider) getVPCRouteTableByTag(routeTableTag *ec2.Tag) ([]*ec2.RouteTable, error) {
```

Comment:
Do we need this function to take in a route table tag now, maybe change to `getCRORouteTable() ([]*ec2.RouteTable, error)` considering we now have `getClusterRouteTables`

Reply:
addressed. Retrieving a tag from inside the function